solace-message-router/azuredeploy.json

```diff
@@ -86,38 +157,71 @@
     "imagePublisher": "OpenLogic",
     "imageOffer": "Centos",
     "OSDiskName": "osdiskfordockersimple",
+    "dataDiskName": "datadiskfordockersimple",
     "nicName": "myVMNicD",
-    "scriptUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/solace-message-router/",
-    "sharedTemplatename": "shared-resources.json",
-    "solaceInstallScriptName": "deploy_vmr.sh",
+    "baseUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/solace-message-router/",
+    "nestedUrl": "[concat(variables('baseUrl'), 'nestedtemplates/')]",
+    "scriptsUrl": "[concat(variables('baseUrl'), 'scripts/')]",
+    "solaceSecurityName": "SolaceSecurity.Template",
+    "securitySharedTemplateName": "security-shared-resources.json",
+    "solaceLoadBalancerName": "SolaceLoadBalancer.Template",
+    "loadBalancerSharedTemplateName": "loadbalancer-shared-resources.json",
+    "solaceWorkspaceName": "SolaceWorkspace.Template",
+    "workspaceEnabledSharedTemplateName": "workspace-enabled-shared-resources.json",
+    "workspaceDisabledSharedTemplateName": "workspace-disabled-shared-resources.json",
+    "solaceInstallScriptName": "deploy_solace.sh",
+    "sempQueryScriptName": "semp_query.sh",
     "publicIPAddressName": "myPublicIPD",
+    "publicIPAddressNameLB": "myLBPublicIPD",
     "publicIPAddressType": "Dynamic",
-    "vmStorageAccountVHDsName": "vhds",
-    "vmName": "SolaceVMR",
     "addressPrefix": "10.0.0.0/16",
     "subnetPrefix": "10.0.0.0/24",
-    "nodeAddressPrefix": "10.0.0.10",
-    "storageAccountType": "Standard_LRS",
     "subnetName": "Subnet",
     "virtualNetworkName": "MyVNETD",
     "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
-    "deploymentModelHighAvailability": {
-      "numberOfNodes": 3
-    },
+    "deploymentModelHighAvailability": { "numberOfNodes": 3 },
     "deploymentModelSingleNode": {
       "numberOfNodes": 1
     },
-    "numberOfInstances": "[variables(concat('deploymentModel', parameters('deploymentModel'))).numberOfNodes]"
+    "numberOfInstances": "[variables(concat('deploymentModel', parameters('deploymentModel'))).numberOfNodes]",
+    "monitorNodeIndex": 2,
+    "lbName" : "myLB",
+    "lbID": "[resourceId('Microsoft.Network/loadBalancers',variables('lbName'))]",
+    "lbPoolName": "solace-ha-group",
+    "lbPoolID": "[concat(variables('lbID'),'/backendAddressPools/',variables('lbPoolName'))]",
```

Comment:
use resourceId for the id

Reply:
Refactored - may required further work, see query for similar comment to frontEndIPConfigID.